cluster/src/main/java/org/apache/iotdb/cluster/server/MetaClusterServer.java

```diff
@@ -56,16 +59,22 @@
   // to register the ClusterMonitor that helps monitoring the cluster
   private RegisterManager registerManager = new RegisterManager();
 
-  public MetaClusterServer() throws QueryProcessException {
+  public MetaClusterServer() throws QueryProcessException, StartupException {
     super();
     member = new MetaGroupMember(protocolFactory, thisNode);
-    // TODO-Cluster#355: check the initial cluster size and refuse to start when the size <
-    //  #replication
+    // check the initial cluster size and refuse to start when the size < quorum
+    int quorum = config.getReplicationNum() / 2 + 1;
+    if (config.getSeedNodeUrls().size() < quorum) {
+      String message = String.format("Seed number less than quorum, seed number: {}, quorum: {}.",
+          config.getSeedNodeUrls().size(), quorum);
+      throw new StartupException(member.getName(), message);
+    }
```

Comment:
This check should only be used when a node tries to form the initial cluster (`startCluster()`). If a node wants to join a cluster, the size of its seeds does not really matter.

Reply:
Thanks for your suggestion. I will change it.